config/configgrpc/bearer_token.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configgrpc
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"regexp"
+
+	"google.golang.org/grpc/credentials"
+)
+
+var _ credentials.PerRPCCredentials = (*PerRPCAuth)(nil)
+
+// PerRPCAuth is a gRPC credentials.PerRPCCredentials implementation that returns an 'authorization' header
+type PerRPCAuth struct {
+	metadata map[string]string
+}
+
+// BearerTokenFromFile builds a new PerRPCAuth with bearer token authentication, reading the token from the specified file
+func BearerTokenFromFile(file string) (*PerRPCAuth, error) {
+	token, err := ioutil.ReadFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't obtain token from file: %w", err)
+	}
+
+	re := regexp.MustCompile(`[\s\n]`)
```

Comment:
I actually thought about making the regex global, but given that this code is executed at most once during the whole lifecycle of the binary, I decided to keep it close to where it's being used. It would also avoid running the code if the feature isn't being used at all.

I personally find the regex more readable, but I have nothing against `bytes.ReplaceAll`, especially given the comment above :-)